framework/src/utils/PetscSupport.C

```diff
@@ -782,32 +825,64 @@ getPetscValidParams()
 }
 
 MultiMooseEnum
-getCommonPetscFlags()
+getCommonSNESFlags()
 {
   return MultiMooseEnum(
-      "-dm_moose_print_embedding -dm_view -ksp_converged_reason -ksp_gmres_modifiedgramschmidt "
-      "-ksp_monitor -ksp_monitor_snes_lg-snes_ksp_ew -ksp_snes_ew -snes_converged_reason "
-      "-snes_ksp -snes_ksp_ew -snes_linesearch_monitor -snes_mf -snes_mf_operator -snes_monitor "
-      "-snes_test_display -snes_view",
+      "-ksp_monitor_snes_lg -snes_ksp_ew -ksp_snes_ew -snes_converged_reason "
+      "-snes_ksp -snes_linesearch_monitor -snes_mf -snes_mf_operator -snes_monitor "
+      "-snes_test_display -snes_view -snes_monitor_cancel",
```

Comment:
yea attempting to keep track of these options is insanity

Reply:
Is there any petsc API that can provide a list of valid options instead of us maintaining these lists?

Reply:
from @balay:

> I don't think there is a single database to query for all valid options.
> 
> The options are distributed across petsc code - and when that part of the code is run - it checks against the input-options base. [i.e. - if that part of the code is -not- run,there is no check for that option]. Also prefix usage in library or use code can change the options.
> 
> Hence './binary -h' prints only the options that are used/available in  the code path of the currently running code. [and it can't print all]

Reply:
I'm going to mark this as resolved for now to allow the merge. Maybe we will think of a better management system in the future. All of these options were hardcoded in `PetscSupport` before I ever touched it, I just further split them into the ksp and snes categories.